### PR TITLE
Fix incremental compilation not working across object references.

### PR DESCRIPTION
**Pull request by cmelchior**
This PR fixes the 2nd bug hit in #1401.

It does it by reworking how the `RealmObjectCompanion` work. Instead of putting the name of a Child relationship into the Parent, the Parent will now instead reference the Child KClass and get the name at runtime.

This approach will work even if @PersistedName on Child is changed, which would not otherwise trigger a recompilation of Parent.

### CODE REVIEW

File: packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt [hunk 1]
```diff
@@ -133,7 +137,7 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
     val classInfoCreateMethod = classInfoClass.lookupCompanionDeclaration<IrSimpleFunction>(CLASS_INFO_CREATE)
 
     private val propertyClass = pluginContext.lookupClassOrThrow(PROPERTY_INFO)
-    val propertyCreateMethod = propertyClass.lookupCompanionDeclaration<IrSimpleFunction>(PROPERTY_INFO_CREATE)
+    val propertyCreateMethod = pluginContext.referenceFunctions(PROPERTY_INFO_CREATE).first()
```
Comment (rorbech) [inline packages/plugin-compiler/src/main/kotlin/io/realm/kotlin/compiler/RealmModelSyntheticPropertiesGeneration.kt:140]: Private?

**Review by rorbech — APPROVED** [review]
Nice fix 🏅

File: packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/schema/CompilerPluginBridgeUtils.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package io.realm.kotlin.internal.schema
```
Comment (clementetb) [inline packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/schema/CompilerPluginBridgeUtils.kt:1]: missing ©

**Review by clementetb — APPROVED** [review]
Nice catch 🎣

**Pull request merged**